qiskit/primitives/containers/estimator_pub.py

```diff
@@ -101,48 +101,62 @@ def precision(self) -> float | None:
         return self._precision
 
     @classmethod
-    def coerce(cls, pub: EstimatorPubLike, precision: float | None = None) -> EstimatorPub:
+    def coerce(
+        cls, pub: EstimatorPubLike, precision: float | None = None, validate: bool = True
+    ) -> EstimatorPub:
         """Coerce :class:`~.EstimatorPubLike` into :class:`~.EstimatorPub`.
 
         Args:
             pub: A compatible object for coercion.
             precision: an optional default precision to use if not
                        already specified by the pub-like object.
+            validate: Whether to validate the returned pub.
 
         Returns:
             An estimator pub.
+
+        Raises:
+            ValueError: If validation fails from a bad value.
+            TypeError: If validation fails from an unexpected type.
         """
         # Validate precision kwarg if provided
         if precision is not None:
             if not isinstance(precision, Real):
                 raise TypeError(f"precision must be a real number, not {type(precision)}.")
             if precision < 0:
                 raise ValueError("precision must be non-negative")
+
         if isinstance(pub, EstimatorPub):
             if pub.precision is None and precision is not None:
                 return cls(
                     circuit=pub.circuit,
                     observables=pub.observables,
                     parameter_values=pub.parameter_values,
                     precision=precision,
-                    validate=False,  # Assume Pub is already validated
+                    validate=validate,
                 )
+            if validate:
+                pub.validate()
```

Comment:
Something to keep in mind: This will typically result in double validation of all pubs if used with primitive args under default behavior (once when a user constructs them, and once when run is called assuming it always goes throug coerce)

Reply:
You're right.